### DISCUSSION TITLE
Educators & GitHub Classroom FAQs 🍎🧑‍🏫

### DISCUSSION THREAD

**Discussion by samus-aran** [community/community#145312] [GitHub Education]
We understand that navigating tools can come with its challenges, so we've compiled the most common questions and answers into one easy-to-access resource. Whether you're just getting started or looking to deepen your expertise, our Educator FAQ is here to provide quick, clear, and practical solutions to support your teaching journey.

> [!IMPORTANT]  
> #### GitHub Classroom is transitioning to partner solutions
> GitHub Classroom will fully transition to partner solutions on **August 28, 2026**. Existing users can continue operating as normal until that date. New users can no longer create classrooms.
> 
> - <a href="#classroom-sunset">What is happening to GitHub Classroom?</a>
> - <a href="#classroom-alternatives">What are the recommended alternative solutions?</a>
> - <a href="#classroom-data-export">How do I export my Classroom data?</a>
> - <a href="#classroom-migration-support">Where can I get migration support?</a>

If you don't see your question, need a little extra guidance, or want to share any feedback please feel free to post it in the community. We'll be keeping this FAQ up-to-date with latest changes and tips. If you have a great tip to share then post in the comments below 👇

So, be sure to bookmark the FAQ, visit regularly, and stay confident knowing you'll always have the current information to make the most out of GitHub Education for your teaching needs.

<details id="quick-links">
<summary> Click Here for Quick Links 🔗🔗 </summary>
<br />

**GitHub Classroom Transition**
- <a href="#classroom-sunset">What is happening to GitHub Classroom?</a>
- <a href="#classroom-alternatives">What are the recommended alternative solutions?</a>
- <a href="#classroom-data-export">How do I export my Classroom data?</a>
- <a href="#classroom-migration-support">Where can I get migration support?</a>

**Educator Benefits & Tools**
- <a href="#verified-teacher"> How do I become a verified GitHub teacher?</a>
- <a href="#teacher-copilot">As a verified educator what Copilot benefits would I receive?</a>
- <a href="#copilot-coding-agent"> When will Copilot Coding Agent be available for verified student/faculty accounts?</a>
- <a href="#free-actions-minutes"> How many free Action minutes would I receive as a verified GitHub teacher?</a>
- <a href="#exceed-minutes"> Would there be any interruptions if I exceed the GitHub Teams' Actions 3000 minutes limit? </a>
- <a href="#limit-student-access"> Can you limit student access to only a selection of repositories on GitHub.com? </a>
- <a href="#created-org-next-steps"> Just created an organization for my school, how do I set up access to my faculty members? </a>
- <a href="#invite-limit"> Can't add more than 50 Students a day to my Organization, what do I do? </a>
- <a href="#LFS-benefits"> As a verified teacher what LFS benefits would I receive? </a>
- <a href="#codespaces-free"> Is Codespaces free for GitHub Education verified teachers?</a>
- <a href="#codespaces-storage"> How does Codespaces Education storage consumption work? </a>
- <a href="#codespaces-increase-allowance"> Our Students want to increase their Codespaces allowance, is that possible? </a>
- <a href="#legal-docs"> What should I do if my school requires a legal or vendor document to be signed by GitHub Education?</a>
- <a href="#upgrade-org"> How do I upgrade my Organization to a Team Plan?</a>
- <a href="#cant-find-my-question">I couldn't find the answer I was looking for.</a>

</details>

---

## GitHub Classroom Transition

<h3 id="classroom-sunset">What is happening to GitHub Classroom?</h3>

[GitHub Classroom](https://classroom.github.com/) is a free tool designed to help educators distribute coding assignments, provide automated feedback, and manage coursework using GitHub repositories.

After operating under maintenance mode for 18 months, we are announcing a transition plan from GitHub Classroom to partner solutions that can more reliably fulfill educators' needs.

**Key dates:**
- **Now**: GitHub Classroom no longer accepts new users. Educators new to the platform cannot create a classroom.
- **Until August 28, 2026**: Existing users can continue to operate as normal (create classrooms, invite students, etc.)
- **August 28, 2026**: GitHub Classroom will fully transition to partner solutions. No users will be able to sign in, and the GitHub Classroom website will be decommissioned.
- **September 4, 2026**: Final Classroom-specific data deletion.

> **Important**: All GitHub user accounts, repositories, and organizations will remain unaffected.

For the full announcement, see the [GitHub Classroom Sunset Discussion](https://gh.io/classroom-sunset).

<h3 id="classroom-alternatives">What are the recommended alternative solutions?</h3>

We've partnered with two organizations that offer the core Classroom workflows (assignment distribution, auto-grading, and GitHub repo integration) plus additional capabilities:

#### Codio

[Codio](https://www.codio.com/) is a hands-on learning platform that combines interactive courseware, browser-based coding environments, and auto-grading in one place. Instructors get free access — no cost to create curriculum, use the migration tool, or explore the full platform. Institutions using Codio for the first time will receive a 50-student credit when purchasing on behalf of students.

Features include:
- LMS integration for single sign-on and grade passback
- Coach AI for in-course student support
- Ability to scale from a single course to department-wide adoption
- Learning and behavioral insights using keystroke-level analytics
- Cheating prevention and plagiarism detection
- A dedicated support team and SLA guarantees

👉 [Learn more about migrating from GitHub Classroom to Codio](https://www.codio.com/github-classroom)

#### Classroom 50

[Classroom 50](https://fifty.foundation/) is a free, open-source GitHub Classroom alternative developed by the Fifty Foundation, GitHub's official open-source partner for educational software tools. It builds on the core Harvard CS50 classroom experience and includes both a command-line interface and a web interface that enables teachers to:

- Distribute programming assignments in repositories
- Configure auto-grading via GitHub Actions
- Review student submissions

Classroom 50 will be available starting **July 1**. Until then, you can [sign up for email updates](https://fifty.foundation/) or [try the pre-release version](https://github.com/foundation50/classroom50/wiki).

<h3 id="classroom-data-export">How do I export my Classroom data?</h3>

GitHub takes the privacy of your data and your students' data seriously. As part of the GitHub Classroom deprecation:

- **GitHub will not transfer your data** to any third-party platform without your direction. Any classroom information you choose to migrate is entirely at your discretion.
- **Your data will remain accessible** through **August 28, 2026**. After that date, Classroom-specific data (classroom/assignment names, tests defined outside of repos, historical data around test runs, submissions, LTI integrated rosters) will be permanently deleted (final deletion September 4, 2026).
- **Export before August 28, 2026**: Use the [Classroom Export Utility](https://github.com/github-education-resources/classroom-export-utility) to export your classroom data to local files.

To learn more about GitHub's privacy practices, see our [Privacy Statement](https://docs.github.com/en/site-policy/privacy-policies/github-general-privacy-statement).

<h3 id="classroom-migration-support">Where can I get migration support?</h3>

The following resources are available to help you through the transition:

- [GitHub Education Support](https://support.github.com/contact/education) – for any specific questions about your GitHub Classroom account
- [GitHub Educator Discussions](https://github.com/community/Global-Campus-Teachers/discussions) – ask questions or share thoughts with the GitHub Education team and fellow educators
- [Codio Book a Consult](https://www.codio.com/consultation) – get a free, personalized consultation tailored to your use case
- [Codio Migration Instructions](https://docs.codio.com/instructors/setupcourses/import-from-github-classroom.html#importfromgithubclassroom) – step-by-step guide and video walkthrough
- [Using GitHub with Codio](https://www.codio.com/on-demand-github-codio-webinar) – on-demand webinar
- [Classroom 50 Teacher Guide](https://github.com/foundation50/classroom50/wiki/Teacher-Guide) – end-to-end walkthrough for instructors
- [Classroom 50 Discussions](https://github.com/foundation50/classroom50/discussions) – post a question or suggestion for the Fifty Foundation Team

> 🎓 **Educator Summit**: GitHub is hosting a virtual Educator Summit on **June 10-12** that will include sessions to support your GitHub Classroom transition. [Learn more and register](https://luma.com/law93emw).

---

## Educator Benefits & Tools

<h3 id="verified-teacher">How do I become a verified GitHub teacher?</h3>

If you're a teacher, you can apply to join GitHub Education and receive access to the resources and benefits of GitHub Education.

For detailed instructions, refer to our documentation on[ applying to GitHub Education as a teacher](https://docs.github.com/en/education/explore-the-benefits-of-teaching-and-learning-with-github-education/github-education-for-teachers/apply-to-github-education-as-a-teacher).

<h3 id="teacher-copilot">As a verified educator what Copilot benefits would I receive?</h3>

Copilot for Students and Teachers is the complimentary offering for successfully verified students and educators.

Check out our FAQs on Copilot for Students and Teachers:
- [What is the difference between Copilot Free and Copilot for Students and Teachers?](https://github.com/orgs/community/discussions/111352#difference-between-copilot-free-and-copilot-for-education)
- [How to enable Copilot Pro for GitHub Education verified students/teachers?](https://github.com/orgs/community/discussions/111352#enable-copilot-education)
- [I'm a verified student/teacher, but my Copilot is not working. What do I do?](https://github.com/orgs/community/discussions/111352#verified-copilot-not-working)

<h3 id="copilot-coding-agent"> When will Copilot Coding Agent be available for verified student/faculty accounts?</h3>

Copilot Coding Agent is available to all verified GitHub Education students and educators. 

<h3 id="free-actions-minutes">How many free Action minutes would I receive as a verified GitHub teacher?</h3>

If you upgrade your [Organization to a Team Plan](https://docs.github.com/en/education/teach-and-learn-with-github-education/apply-for-an-educator-or-researcher-discount#upgrading-your-organization) for free by becoming a [verified GitHub Education educator](https://education.github.com/discount_requests/application), you'll be able to get 3,000 actions minutes instead of 2,000.

Here is another link showing the amount of minutes for the Team plan:

[https://docs.github.com/en/get-started/learning-about-github/githubs-products#github-team](https://docs.github.com/en/get-started/learning-about-github/githubs-products#github-team)

<h3 id="exceed-minutes">What happens if I exceed the GitHub Teams' Actions 3,000 minutes limit?</h3>

Set your [spending limit to "$0.00"](https://docs.github.com/en/enterprise-cloud@latest/billing/managing-billing-for-your-products/managing-billing-for-github-actions/managing-your-spending-limit-for-github-actions#managing-the-spending-limit-for-github-actions-for-your-personal-account) for GitHub Actions & Packages on your organization account. The minutes reset on the 1st of each month at midnight UTC. 

You can learn more about billing for any overages here:

[https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates)

<h3 id="limit-student-access">Can I limit student access to only a selection of repositories on GitHub.com?</h3>

GitHub doesn't have an inbuilt way of preventing access to certain content that would help a facilitator or school limit the breadth of access on GitHub.com.

Generally for institution-owned devices, these options are best configured by your IT team, for example to prevent certain code from running, or from loading certain web pages deemed unsuitable. Given that a determined student may be able to access the content elsewhere, or on another device and transfer it - these kinds of restrictions are always at best a guide rather than true protection that can be achieved at the device level.

Approved educational institutions can deploy a locally hosted GitHub Enterprise Server instance for free, see [https://github.com/education/schools](https://github.com/education/schools). A dedicated local instance can be configured in such a way that only repositories on the instance can be accessed, although this instance would **not support** GitHub Codespaces.

Alternatively if you were wanting to create a curated group of repositories that you could share with others, this can be achieved via the GitHub lists feature, for more information, read [Creating a list ](https://docs.github.com/get-started/exploring-projects-on-github/saving-repositories-with-stars#creating-a-list)in the GitHub Docs.

<h3 id="created-org-next-steps">I just created an organization for my school, how do I set up access to my faculty members?</h3>

Check here for information on how to [provide access restrictions to an organization.](https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization) You can also [create specific Teams](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams) and limit their access.

An alternative solution would be for your school to apply to become a [GitHub Education partner](https://education.github.com/partners/schools) and have a school Admin who manages an Enterprise account. This might be helpful in case you should leave your institution or if you might lapse on renewing your benefits, as such changes to your individual organization would affect all the users.

<h3 id="invite-limit">I can't add more than 50 Students a day to my Organization, what do I do?</h3>

To prevent abuse, you can only create 50 organization invitations within a 24-hour period. If your organization is more than one month old or on a paid plan, the limit is 500 invitations per 24 hour period. As a verified faculty member you can upgrade an organization to a GitHub team which will increase your invitation limit to that of a paid plan. Check out more on how to invite users to your organization in [our documentation](https://docs.github.com/en/organizations/managing-membership-in-your-organization/inviting-users-to-join-your-organization).

<h3 id="LFS-benefits">As a verified teacher what LFS benefits would I receive?</h3>

The academic discount granted to your GitHub organization includes support for GitHub Large File Storage (LFS).

To add LFS data packs you will need to adjust some settings in your organization account by following instructions from this help article (the article mentions billing but as you go through the process there will be no cost):

[Upgrading Git Large File Storage](https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/upgrading-git-large-file-storage)

The following guide should help with actual usage of GitHub LFS:

[Managing large files](https://docs.github.com/en/repositories/working-with-files/managing-large-files)

<h3 id="codespaces-free"> Is Codespaces free for GitHub Education verified teachers? </h3>

As a verified faculty, you have access to free upgrades of organizations to GitHub Teams.
The current [GitHub Team plan](https://github.com/pricing#compare-features) for organizations does not include free use of GitHub Codespaces. More details on Codespaces pricing can be found [here](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#about-github-codespaces-pricing).

GitHub Free for personal accounts includes 120 core hours and 15 GB storage per month for Codespace.
 
Students are welcome to visit GitHub Education to apply for academic benefits.
Getting verified will give them access to Student Developer Pack, including GitHub Pro, Copilot Pro, and 180 core hours and 20 GB storage per month for Codespace.
Additional details on monthly included storage and core hours for personal accounts can be found here.
 

Organizations can choose whether codespaces created from their repositories will be user-owned or organization-owned. See [Choosing who owns and pays for codespaces in your organization.](https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/choosing-who-owns-and-pays-for-codespaces-in-your-organization) An organization pays for a codespace if all the following things are true:
 
The organization has chosen for codespaces to be organization-owned.
The organization has set a non-zero spending limit for GitHub Codespaces.
The codespace is created from one of the organization's repositories, or from a fork of one of the organization's repositories. This includes both public and private repositories.
The user creating the codespace is a member or collaborator of the organization, and the organization has enabled GitHub Codespaces for this user. This can include all members and collaborators if the organization has chosen to enable Codespaces for all users. If Codespaces isn't enabled for a user, they can still create codespaces from public repositories in the organization, but the user will pay for these codespaces.

<h3 id="codespaces-storage"> How does Codespaces Education storage consumption work? </h3>

Codespaces continue to use storage even when they are not actively in use. To manage storage effectively, we suggest deleting any old Codespaces that are no longer needed. Note that even after deletion, Codespaces will still contribute to your storage usage until the current billing cycle concludes.

**Monthly Usage Reset**
Usage automatically resets each month on your account's billing date.

**Accessing Detailed Usage Reports**
You can download a detailed usage report from the billing page:

* **For Personal Accounts**:[ GitHub Billing Page](https://github.com/settings/billing)
* **Step by Step Instructions included**:[ Viewing Your GitHub Actions Usage](https://docs.github.com/en/billing/managing-billing-for-github-actions/viewing-your-github-actions-usage)

**Managing Prebuilds**
To avoid additional costs and save space, consider removing Prebuilds as outlined here:
[Deleting a Prebuild Configuration](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/managing-prebuilds#deleting-a-prebuild-configuration)

**Learn More** 
Read more about how storage and pricing for Codespaces are calculated:
[About Billing for GitHub Codespaces](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces)

<h3 id="codespaces-increase-allowance"> Our Students want to increase their Codespaces allowance, is that possible? </h3>

Verified students can enjoy free access to GitHub Codespaces, with up to 180 core hours per month for their personal accounts. This monthly allowance of storage and usage is the same as what's included with GitHub Pro accounts. For more details, check out About billing for GitHub Codespaces.

Additional details on monthly included storage and core hours for personal accounts can be found here.
 
Users are notified by email when they have used 75%, 90%, and 100% of their included quotas.
When a personal account has used all of either the included storage or compute usage (whichever is reached first), and has no spending limit configured, use of GitHub Codespaces will be blocked. A payment method and a spending limit must be set up to continue using GitHub Codespaces during the current billing month. At the beginning of the next monthly billing cycle the included usage is reset. Storage will not be billed while use of GitHub Codespaces is blocked.

<h3 id="legal-docs"> What should I do if my school requires a legal or vendor document to be signed by GitHub Education? </h3>

Since GitHub Education offers free products, we don't have the resources to review external documents. However, if your school is interested in exploring a paid product, we'd be happy to forward your inquiry to our sales team. Our sales team can help explore options that best fit your school's needs and answer any questions you may have.

<h3 id="upgrade-org"> How do I upgrade my Organization to a Team Plan? </h3>

Verified faculty can upgrade any GitHub organization being used for academic purposes to [GitHub Team](https://github.com/pricing) with a single click.

If you have organizations you can upgrade you can visit [GitHub Education dashboard](https://education.github.com/globalcampus/teacher) and find "Upgrade your academic organizations" near the top.

<img src="https://github.com/user-attachments/assets/8a41cd72-fb1e-4989-8bcc-95d2c08e306d" width="50%" />

Click the **Upgrade to GitHub Team** button and then click the Upgrade button next to the organization name to upgrade it for free to GitHub Team.

<h3 id="cant-find-my-question"> I couldn't find the answer I was looking for </h3>

We continuously update these FAQs but if you are unable to find the answer to your question please raise a discussion on the [GitHub Education community](https://github.com/orgs/community/discussions/categories/github-education) and the Community team will be happy to assist 🌟

**Comment by M-tech-cmd** [comment]
For more detailed information, you can visit the GitHub Classroom FAQ page 

(https://docs.github.com/en/education/manage-coursework-with-github-classroom/get-started-with-github-classroom): [GitHub Classroom FAQ](https://docs.github.com/en/education/manage-coursework-with-github-classroom/get-started-with-github-classroom)

**Comment by Apashash** [comment]
Good

**Comment by jennirinker** [comment]
A suggestion for an addition: as a new teacher, it wasn't clear to me whether the "organization" should be my _institute_ or my _class_. I know this is up to the user, but it might be good to put a suggested default. In my case, I've decided to make the "organization" actually the class, not my institute. This is because there are otherwise the org page on GitHub got cluttered with hundreds of student repos from different courses. But perhaps there is a better way forward...

**Comment by rommelcarneiro** [comment]
@samus-aran, we forgot to check the "I consent to having my project featured through interviews, blog posts, and/or live events if selected." option in the form, but it was an accident. How can we change this information in our submission? Can you make this change?

**Comment by metinmeki** [comment]
Wow !

**Comment by kingma-sbw** [comment]
To the FAQs could you add how to deal with the "Repository Access Issue" error message? Having multiple students not being able to start on an assignment is a horrible experience, certainly when there is a deadline.
<img width="451" height="222" alt="image" src="https://github.com/user-attachments/assets/d34f483c-91ab-421e-9d6d-d6d4ffd7639f" />

**Comment by SMUGSterling** [comment]
Can you provide guidance on what qualifies as "A Teacher"

I ask this NOT to game the system, but because I have folks like:

- Faculty that aren't *currently* teaching but are advising students and assisting student research
- Librarians
- Dedicated research support staff that don't teach "for credit courses" but who DO teach things like Professional Development, GitHub, HPC, etc.

Are these folks allowed to apply for GitHub Educator Licenses?

**Comment by etLLMhome** [comment]
What a laughably miserable experience this has been! 

I happened to sign up for github with my edu email account, and then you people started showing me banners asking me to sign up for Github Education. Roughly 5 applications later, I have uploaded pictures of myself, my staff ID card, my employment verification letter, my payslips, my grandmother's birth certificate, and I am still being rejected. 

Thankfully I eventually came to check this FAQ, to realise that this performative corporate social responsibility project is being terminated anyway. It would be GREAT if you stopped advertising it to people and wasting their time. Thanks!

**Comment by dagilleland** [comment]
Looks like GitHub Classroom's sunset is triggering a lot of in-house replacements, ranging from online services to ad-hoc scripts. I've also created my own - a ***desktop application*** - for your consideration, and I just released the first public beta.

See [**✨ CodeHub Classroom**](https://codehub-classroom.github.io/)

<img width="895" height="913" alt="image" src="https://github.com/user-attachments/assets/749fd6a8-e5c5-4dc6-a84a-f478297e5460" />

I've been using GitHub Classroom since 2013, so I'm trying to bring in my "wish list" features from over the years.

Let me know what you think!

**Comment by evenicholson12** [comment]
This is a useful resource for educators managing coding assignments and student coursework, especially with the recent transition away from GitHub Classroom. Clear guidance on assignment distribution, feedback, grading, and alternative platforms can make the learning process much easier for both teachers and students. The Academic Papers UK also provides **[Assignment Help](https://www.theacademicpapers.co.uk/assignment-writing-services/
)** to support students with research, structure, and academic writing.
